src/daemon/rpmostreed-utils.cxx

```diff
@@ -609,3 +609,42 @@ rpmostreed_parse_revision (const char  *revision,
 out:
   return ret;
 }
+
+/* Throw an error if there exists systemd inhibitor locks in `block` mode only.
+ * Note: systemd 248 provides a `--check-inhibitors` option, but it also checks
+ * for inhibitors in `delay` mode, which isn't what we want. */
+gboolean
+check_sd_inhibitor_locks (GDBusConnection *connection,
+                          GCancellable    *cancellable,
+                          GError         **error)
+{
+  g_autoptr(GVariant) inhibitors_array_tuple = 
+    g_dbus_connection_call_sync (connection, "org.freedesktop.login1", "/org/freedesktop/login1",
+                                 "org.freedesktop.login1.Manager", "ListInhibitors",
+                                 NULL, (const GVariantType*)"(a(ssssuu))",
+                                 G_DBUS_CALL_FLAGS_NONE, -1, cancellable, error);
+  if (!inhibitors_array_tuple)
+    return FALSE;
+  else if (g_variant_n_children (inhibitors_array_tuple) < 1)
+    return glnx_throw (error, "LoadInhibitors returned empty tuple");
+  g_autoptr(GVariant) inhibitors_array =
+    g_variant_get_child_value (inhibitors_array_tuple, 0);
+
+  char *what;
+  char *mode;
+  int owner;
+  int num_block_inhibitors = 0;
+  GVariantIter viter;
+  g_variant_iter_init (&viter, inhibitors_array);
+  while (g_variant_iter_loop (&viter, "(ssssuu)", &what, NULL, NULL, &mode, &owner, NULL))
+    {
+      // Only consider shutdown inhibitors in `block` mode owned by root.
```

Comment:
systemd already has policykit checks on adding blocking inhibitors - I see no reason to ignore inhibitors set up by non-root if they were authorized.  So let's just drop this `owner == 0` check?

For example:
```
[core@cosa-devsh ~]$ systemd-inhibit bash
Failed to inhibit: Access denied
[core@cosa-devsh ~]$
```